wolfcrypt/src/asn.c

```diff
@@ -5572,7 +5572,7 @@ static int SetRsaPublicKey(byte* output, RsaKey* key,
     n[0] = ASN_INTEGER;
     nSz  = SetLength(rawLen, n + 1) + 1;  /* int tag */
 
-    if ( (nSz + rawLen) < MAX_RSA_INT_SZ) {
+    if ( (nSz + rawLen) <= MAX_RSA_INT_SZ) {
```

Comment:
Yes, this includes the case where leadingBit is 1.

In case of 4096-bit key:

(mp_unsigned_bin_size() + leadingBit) == rawLen == 513 bytes
ASN INTEGER encodes as 4 bytes (TAG + 3 length bytes) == nSz == 4 bytes

This makes (nSz + rawLen) == MAX_RSA_INT_SZ == 517